hemework25-datestr.py

```diff
@@ -0,0 +1,160 @@
+#TASK1
+from functools import wraps
+def check_full(method):
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        if self.is_full():
+            raise ValueError("Storage is full.")
+        return method(self, *args, **kwargs)
+    return wrapper
+
+def check_empty(method):
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        if self.is_empty():
+            raise ValueError("Storage is empty.")
+        return method(self, *args, **kwargs)
+    return wrapper
+
+class BaseSequence:
+    """A base sequence class."""
+    def __init__(self, initial_items: list, capacity=10):
+        self._storage = initial_items
+        self._capacity = capacity
+
+    @check_empty
+    def put(self, element):
+        """Put an element into the sequence."""
+        if self.is_full():
+            raise ValueError("Storage is full.")
+        self._storage.append(element)
+
+    def is_empty(self):
+        """Check if the sequence is empty."""
+        return not bool(self._storage)
+
+    def is_full(self):
+        """Check if the sequence is full."""
+        return len(self._storage) == self._capacity
+
+
+class Stack(BaseSequence):
+    """A Last-In-First-Out (LIFO) data structure."""
+    @check_empty
+    def pop(self):
+        """Remove and return the last element."""
+        if self.is_empty():
+            raise ValueError("Storage is empty.")
+        return self._storage.pop()
+
+    def peek(self):
+        """Return the last element without removing it."""
+        if self.is_empty():
+            raise ValueError("Storage is empty.")
+        return self._storage[-1]
+
+class Queue(BaseSequence):
+    """A First-In-First-Out (FIFO) data structure."""
+    def get_beginning(self):
+        """Remove and return the first element."""
+        if self.is_empty():
+            raise ValueError("Storage is empty.")
+        return self._storage.pop(0)
+
+
+class Dequeue(Queue, Stack):
+    """A sequence that allows insertion and removal from both ends."""
+    @check_full
+    def put_beginning(self, element):
+        """Insert an element at the beginning of the dequeue."""
+        if len(self._storage) >= self._capacity:
+            raise ValueError("Storage is full.")
+        self._storage.insert(0, element)
+
+def test_sequence_operations():
+    """Test various operations on the sequences."""
+    # Test an empty stack
+    stack = Stack([])
+    try:
+        stack.pop()  # Should raise ValueError
```

Comment:
In case when `stack.pop()` doesn't raise a ValueError by some reason - test still passes (that's called false positive). If you _expect_ an error - you should use `with pytest.raises` (google it please).